code/game/machinery/biogenerator.dm

```diff
@@ -329,6 +329,18 @@
 	speed_coefficient = 2/manipcount
 	biomass_coefficient = 3*lasercount
 
+/obj/machinery/biogenerator/AltClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && user.dexterity_check())
+		eject_produce()
+		return
+	return ..()
+
+/obj/machinery/biogenerator/CtrlClick(mob/user)
+	if(!user.incapacitated() && Adjacent(user) && beaker && user.dexterity_check())
```

Comment:
Since the biogenerator has `WRENCHMOVE`, make this not pass if it's unbolted so you can still use control-click to pull it.